actions/get-latest-workflow-artifact/src/main.ts

```diff
@@ -32,50 +33,96 @@ async function main() {
   });
 
   const headSha = data.head.sha;
-  // Now let's get all workflows associated with this sha.
-  const { data: runs } = await client.rest.actions.listWorkflowRuns({
-    repo: repoName,
-    owner: repoOwner,
-    head_sha: headSha,
-    workflow_id: workflowId,
-    status: "completed",
-  });
 
-  if (runs.workflow_runs.length <= 0) {
-    throw new Error(`No workflow runs found for sha ${headSha}`);
-  }
+  // Now let's get all workflows associated with this sha:
+  // We start with in-progress ones if those are to be considered.
+  let found = null;
```

Comment:
needs to be typed, otherwise the assignments further down aren't valid, since `found` will have type `null`

```suggestion
  let found: { artifact: GetArtifactResponse; run: WorkflowRun } | null = null;
```